core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -138,79 +147,192 @@ private fun computePaths(
     }
 
     // Handling errors
-    // Check if pathfinding failed due to constraints
+    // Check if pathfinding failed due to incompatible constraints
+    val elapsedSeconds = Duration.between(start, Instant.now()).toSeconds()
+    val incompatibleConstraintsResponse =
+        buildIncompatibleConstraintsResponse(
+            infra,
+            waypoints,
+            constraints,
+            remainingDistanceEstimators,
+            timeout?.minus(elapsedSeconds)
+        )
+    if (incompatibleConstraintsResponse != null) {
+        throw NoPathFoundException(incompatibleConstraintsResponse)
+    }
+    // It didn’t fail due to an incompatible constraint, no path exists
+    throw NoPathFoundException(NotFoundInBlocks(listOf(), Length(0.meters)))
+}
+
+private fun buildIncompatibleConstraintsResponse(
+    infra: FullInfra,
+    waypoints: ArrayList<Collection<PathfindingEdgeLocationId<Block>>>,
+    constraints: List<PathfindingConstraint<Block>>,
+    remainingDistanceEstimators: List<AStarHeuristicId<Block>>,
+    timeout: Double?
```

Comment:
Not sure we need the timeout. Timeout is used to prevent the algorithm to compute for too much time when it doesn't find a solution. Since we've arrived here, it means the constrained pathfinding has already taken less than 2min. As a result, there's no reason the unconstrained pathfinding should take more than 2min.

Reply:
> Since we've arrived here, it means the constrained pathfinding has already taken less than 2min. As a result, there's no reason the unconstrained pathfinding should take more than 2min.

It can, the unconstrained pathfinding has more edges to explore.

Reply:
After first discussions: we want to be able to distinguish a timeout using initial (real) constraints and a timeout unconstrained if we have an "unconstrained" timeout.
Now the question is (@bloussou ?): What timeout do we want on unconstrained PF (launched to "explain" failure) ?
1. shared with the constrained PF
2. equal to the constrained PF
3. no timeout

Reply:
I think it should be shared no ? Is it triggered for each failing PF ?

Reply:
Yes. For each failing PF, the unconstrained PF is triggered, to find out why it failed (ex: due to electrification constraints), the explanation is returned and will be displayed in the front end of the app (back end part of this feature is this PR).
If your question was if the timeout was triggered for every failing PF, the answer is no, it depends.

Reply:
We may not care about explaining PF failure (RS constraints) when using the timeout (BASIC import).
It would let more freedom and maybe aim at a quicker fail, so a mutualized timeout?

Reply:
okay lets go for a mutualized timeout I would say ?

Reply:
OK, so the timeout handling didn't change, but a dedicated error type was added in https://github.com/OpenRailAssociation/osrd/pull/7953/commits/9a63a73b0cfadf86020072793530da2eb6b8e6e5